docs/languages/csharp/expression-trees.md

```diff
@@ -0,0 +1,65 @@
+﻿# Expression Trees
+
+If you have used LINQ, you have experience with a rich library
+where `Func` and `Func<T>` are part of the API. (If you are not familiar
+with LINQ, you probably want to read [that tutorial](linq.md) and
+the tutorial on [lambda expressions](lambda-expressions.md) before this one.)
+*Expression Trees* provide richer interaction with the arguments that
+are functions.
+
+Using a typical LINQ query, and function arguments are represented
+by a delegate the compiler creates.
+
+When you want to have a richer interaction, you need to use *Expression Trees*.
+Expression Trees represent code as a structure that you can examine,
+modify, or execute. These tools give you the power to manipulate code during
+runtime. You can write code that examines running algorithms, or injects new
+capabilities. In more advanced scenarios, you can modify running algorithms,
+and even translate C# expressions into another form for execution in another
+environment.
+
+You've likely already written code that uses Expression Trees. Entity Framework's
```

Comment:
added locally. I linked here: http://docs.efproject.net/en/latest/  would that be best?